gcp_variant_transforms/vcf_to_bq_common.py

```diff
@@ -123,16 +123,10 @@ def write_headers(merged_header, file_path):
 def _raise_error_on_unrecognized_flags(pipeline_args):
   # type: (List[str]) -> None
   """Raises an error if there are unrecognized flags."""
-  options = pipeline_options.PipelineOptions(pipeline_args).get_all_options()
-  for flag in _get_flag_names(pipeline_args):
-    # Cannot use exact match since the argparse allows long options to be
-    # abbreviated to a prefix.
-    if not any(option.startswith(flag) for option in options):
-      raise ValueError('The flag {} is unrecognized.'.format(flag))
-
-
-def _get_flag_names(pipeline_args):
-  # type: (List[str]) -> List[str]
-  """Returns a list of flag names that starts with `COMMAND_LINE_ARG_PREFIX`."""
-  return [arg.lstrip(_COMMAND_LINE_ARG_PREFIX) for arg in pipeline_args
-          if arg.startswith(_COMMAND_LINE_ARG_PREFIX)]
+  parser = argparse.ArgumentParser()
+  for cls in pipeline_options.PipelineOptions.__subclasses__():
+    if '_add_argparse_args' in cls.__dict__:
+      cls._add_argparse_args(parser)
+  _, unknown = parser.parse_known_args(pipeline_args)
+  if unknown:
+    raise ValueError('The flag {} is unrecognized.'.format(unknown))
```

Comment:
nit: s/flag/flag(s) as there can be multiple ones...I recommend rephrasing as `Unrecognized flag(s): {}`

Reply:
Thanks! That's better.